tests/unit/test_datasets.py

```diff
@@ -0,0 +1,92 @@
+import io
+from unittest.mock import Mock, patch
+from zipfile import ZipFile
+
+import botocore
+
+from sdgym.datasets import download_dataset
+
+
+class AnyConfigWith:
+    """AnyConfigWith matches any s3 config with the specified signature version."""
+    def __init__(self, signature_version):
+        self.signature_version = signature_version
+
+    def __eq__(self, other):
+        return self.signature_version == other.signature_version
+
+
+@patch('sdgym.datasets.boto3')
+def test_download_dataset_public_bucket(boto3_mock, tmpdir):
+    """Test downloading datasets from a public bucket."""
+    # setup
+    dataset = 'my_dataset'
+    bucket = 'my_bucket'
+    bytesio = io.BytesIO()
+    with ZipFile(bytesio, mode='w') as zf:
+        zf.writestr(dataset, 'test_content')
```

Comment:
I would add blank lines above and below the `with` block, to separate the steps more clearly.